### PR TITLE
レビュー機能のデザインと修正を行いました

### PR DESCRIPTION
**Pull request by yamada-kensyu**
・詳細画面のレビューのデザインを変更しました
・一覧画面にレビューが表示されるようにしました
・レビュー投稿画面とレビュー更新画面のデザインを変更しました
・SQL文にレビューとユーザを追加しました
・Itemオブジェクトのレビュー平均を計算するメソッドが、正しくなかったので修正しました

### CODE REVIEW

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 import jp.co.example.ecommerce_a.domain.Item;
 import jp.co.example.ecommerce_a.repository.ItemRepository;
+import jp.co.example.ecommerce_a.repository.ReviewRepository;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:11]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'jp.co.example.ecommerce_a.repository.ReviewRepository' import. Should be before 'org.springframework.transaction.annotation.Transactional'.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 2]
```diff
@@ -21,6 +22,8 @@ public class ShowItemListService {
 
 	@Autowired
 	private ItemRepository itemRepository;
+	@Autowired
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:25]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:25]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'member def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 3]
```diff
@@ -21,6 +22,8 @@ public class ShowItemListService {
...
+	private ReviewRepository reviewRepository;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:26]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 4]
```diff
@@ -32,6 +35,9 @@ public class ShowItemListService {
 	public List<Item> searchByName(String name, String sortNum) {
 		List<Item> itemList = null;
 		itemList = itemRepository.findByNameWithSort(sortNum, name);
+		for (Item item : itemList) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:38]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 5]
```diff
@@ -32,6 +35,9 @@ public class ShowItemListService {
...
+			item.setReviewList(reviewRepository.findByItemId(item.getId()));
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:39]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 6]
```diff
@@ -32,6 +35,9 @@ public class ShowItemListService {
...
+		}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:40]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for rcurly' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 7]
```diff
@@ -50,6 +56,9 @@ public List<Item> narrowingDown(Integer targetAge, Integer genre, Integer priceR
 		} else {
 			itemList = itemRepository.findByTargetAgeGenreAndPriceRange(targetAge, genre, priceRange);
 		}
+		for (Item item : itemList) {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:59]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:59]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 8]
```diff
@@ -50,6 +56,9 @@ public List<Item> narrowingDown(Integer targetAge, Integer genre, Integer priceR
...
+			item.setReviewList(reviewRepository.findByItemId(item.getId()));
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:60]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for' child has incorrect indentation level 24, expected level should be 6.

File: src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java [hunk 9]
```diff
@@ -50,6 +56,9 @@ public List<Item> narrowingDown(Integer targetAge, Integer genre, Integer priceR
...
+		}
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:61]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/service/ShowItemListService.java:61]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'for rcurly' has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/domain/Item.java [hunk 10]
```diff
@@ -79,15 +79,16 @@ public Item(Integer id, String name, String description, Integer priceM, Integer
 	 * 
 	 * @return　レビューの平均
 	 */
-	public float getCalcAveRating() {
-		float aveRating = 0;
+	public double getCalcAveRating() {
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:82]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def modifier' has incorrect indentation level 8, expected level should be 2.

File: src/main/java/jp/co/example/ecommerce_a/domain/Item.java [hunk 11]
```diff
@@ -79,15 +79,16 @@ public Item(Integer id, String name, String description, Integer priceM, Integer
...
+		double aveRating = 0;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:83]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:83]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'method def' child has incorrect indentation level 16, expected level should be 4.

File: src/main/java/jp/co/example/ecommerce_a/domain/Item.java [hunk 12]
```diff
@@ -79,15 +79,16 @@ public Item(Integer id, String name, String description, Integer priceM, Integer
...
 		int totalRating = 0;
 		for (Review review : reviewList) {
 			totalRating += review.getRating();
 		}
 		// 平均を小数第二位で四捨五入する
 		if (reviewList.size()!=0) {
-			aveRating = Math.round(totalRating / reviewList.size() * 10)/10;
+			double rawAveRating=  (double)totalRating / reviewList.size();
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:90]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:90]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:90]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '=' is not preceded with whitespace.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:90]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'typecast' is not followed by whitespace.

File: src/main/java/jp/co/example/ecommerce_a/domain/Item.java [hunk 13]
```diff
@@ -79,15 +79,16 @@ public Item(Integer id, String name, String description, Integer priceM, Integer
...
+			aveRating = ((double)Math.round(rawAveRating * 10))/10;
```
Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:91]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.FileTabCharacterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line contains a tab character.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:91]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' child has incorrect indentation level 24, expected level should be 6.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:91]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'typecast' is not followed by whitespace.

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:91]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '/' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3)

Comment (SekiMoemi) [inline src/main/java/jp/co/example/ecommerce_a/domain/Item.java:91]: ⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '/' is not preceded with whitespace.

**Review by quangtung98 — APPROVED** [review]
問題がなさそうと思いますので、承認します。
作成ありがとうございます

**Pull request merged**